src/main/java/net/imglib2/img/AbstractNativeImg.java

```diff
@@ -77,4 +78,13 @@ public T createLinkedType()
 			return null;
 		}
 	}
+
+	@Override
+	public T getType() {
+		try {
+			return linkedType.createVariable();
+		} catch ( final NullPointerException e ) {
+			return super.getType();
+		}
+	}
```

Comment:
This could move into the `NativeImage` interface as
```
	@Override
	public default T getType() {
		return createLinkedType().createVariable();
	}
```
Is there a performance penalty to creating a linked type vs just `createVariable()`? Is this `null` safe or should we add a `null` check ?